build/targets/KoreBuild.SolutionBuild.targets

```diff
@@ -41,27 +45,60 @@ as they are subject to change.
 -->
 
   <Target Name="CleanArtifacts">
-    <!-- intentionally BuildDir, not ArtifactDir -->
-    <RemoveDir Directories="$(BuildDir)" />
+    <ItemGroup>
+      <_FilesToDelete Include="$(ArtifactsDir)**\*" Exclude="$(ArtifactsDir)msbuild\**\*" />
+    </ItemGroup>
+    <Delete Files="@(_FilesToDelete)" />
+  </Target>
+
+  <Target Name="_PrepareOutputPaths">
+    <MakeDir Directories="$(ArtifactsDir);$(BuildDir)" />
+  </Target>
+
+  <Target Name="_ResolveCommitHash" Condition="'$(CommitHash)'==''">
+    <PropertyGroup>
+      <CommitHash Condition="'$(APPVEYOR_REPO_COMMIT)' != ''">$(APPVEYOR_REPO_COMMIT)</CommitHash>
+      <CommitHash Condition="'$(TRAVIS_COMMIT)' != ''">$(TRAVIS_COMMIT)</CommitHash>
+    </PropertyGroup>
+
+    <GetGitCommitInfo WorkingDirectory="$(RepositoryRoot)"
+                      Condition="'$(CommitHash)' == ''"
+                      ContinueOnError="WarnAndContinue">
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+      <Output TaskParameter="Branch" PropertyName="GitBranch" />
```

Comment:
Where is `$(GitBranch)` used? Adding it to `@(SolutionProperties)` doesn't mean much if it isn't read somewhere else.